TECHMANIA/Assets/Scripts/Components/Editor Scene/EditorTopBar.cs

```diff
@@ -27,6 +27,19 @@ private void OnDisable()
     public void Save()
     {
         EditorContext.Save();
+#if !UNITY_IOS && !UNITY_ANDROID
```

Comment:
There is too much macro-guarded code around various components and this worries me. How about this: add a method `bool DiscordController.SupportedOnCurrentPlatform()`, which returns a boolean depending on platform, and then use `if (DiscordController.SupportedOnCurrentPlatform())` instead of macro guards in these components.